spec/presenters/api/v2/measures/measure_presenter_spec.rb

```diff
@@ -10,6 +10,38 @@
     end
   end
 
+  context 'when measure has a generating regulation' do
+    let(:measure) { create(:measure, :with_base_regulation) }
+
+    describe '#measure_generating_legal_act' do
+      it 'returns the MeasureLegalActPresenter' do
+        expect(presenter.measure_generating_legal_act).to be_instance_of(Api::V2::Measures::MeasureLegalActPresenter)
+      end
+    end
+
+    describe '#measure_generating_legal_act_id' do
+      it 'has correct ID' do
+        expect(presenter.measure_generating_legal_act_id).to eq(measure.measure_generating_regulation_id)
+      end
+    end
+  end
+
+  context 'when measure has a justification_legal_act' do
+    let(:measure) { create(:measure, :with_justification_regulation) }
+
+    describe '#measure_generating_legal_act' do
+      it 'returns the MeasureLegalActPresenter' do
+        expect(presenter.justification_legal_act).to be_instance_of(Api::V2::Measures::MeasureLegalActPresenter)
```

Comment:
Can this also have an additional spec for comparing the id as well as the type

Reply:
I've added this:
`expect(presenter.justification_legal_act.base_regulation_id).to eq(measure.justification_regulation_id)`
to check the `justification_legal_act` is the correct one.

The following one checks the `measure_generating_legal_act_id` (which is a delegation).